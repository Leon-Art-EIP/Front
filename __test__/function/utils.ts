import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
