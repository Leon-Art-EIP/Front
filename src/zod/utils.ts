export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
