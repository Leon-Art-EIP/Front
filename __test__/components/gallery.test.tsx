import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Gallery from "../../src/components/gallery";

test("Gallery snapshot", () => {
  testSnapshot(<Gallery redirectText="Test" redirectUrl="https://test.com" />);
});
