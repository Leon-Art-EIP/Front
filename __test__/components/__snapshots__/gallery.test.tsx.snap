// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gallery snapshot 1`] = `
<div>
  <div
    className="flex justify-center"
  >
    <label
      className="text-6xl font-extrabold mt-7"
    >
      <span
        className="text-primary cursor-default"
      >
        Leon
      </span>
      <span
        className="text-tertiary cursor-default"
      >
        '
        Art
      </span>
    </label>
    <a
      href="https://test.com"
    >
      <button
        className="absolute mt-9 right-7 rounded-[30px] shadow-md px-4 py-3 border border-secondary text-xl font-extrabold text-primary cursor-pointer hover:bg-secondary-hover hover:scale-105 transition-transform ease-out"
      >
        Test
      </button>
    </a>
  </div>
  <div
    className="flex justify-evenly mt-10"
  >
    <label
      className="text-4xl font-bold text-tertiary"
    >
      Artistes
    </label>
    <label
      className="text-4xl font-bold text-tertiary"
    >
      Cat√©gories
    </label>
    <label
      className="text-4xl font-bold text-tertiary"
    >
      Support
    </label>
    <label
      className="text-4xl font-bold text-tertiary"
    >
      Couleurs
    </label>
  </div>
  <div
    className="flex justify-center pb-10"
  >
    <div
      className="bg-background grid grid-cols-4 gap-8 place-items-center justify-center py-4 px-4"
    />
  </div>
</div>
`;
